File: Data Structures & Algorithms/products-of-array-discluding-self/submission-4.cs
public class Solution {
    public int[] ProductExceptSelf(int[] nums) {
        var totalProduct =0;
        bool isZero = false;
        var output = new int[nums.Length];
        foreach(int num in nums)
        {
            if(num == 0)
            {
                if(isZero)
                {
                    totalProduct = 0;
                }
                else
                {
                    isZero = true;
                }
            }
            else
            {
                if(totalProduct == 0 && !isZero)
                {
                    totalProduct++;
                }
                totalProduct *= num;
            }
        }
        for(int i = 0; i < nums.Length; i++)
        {
            if(isZero && nums[i] != 0)
            {
                output[i] = 0;
            }
            else if(nums[i] == 0)
            {
                output[i] = totalProduct;
            }
            else
            {
                output[i] = totalProduct / nums[i];
            }
        }
        return output;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s, longest=set(nums), 0
        for n in s:
            if n - 1 in s:
                continue
            l = 1
            while n + l in s:
                l += 1
            longest = max(l, longest)
        return longest
            
                